services/slam/builtin/builtin.go

```diff
@@ -868,17 +874,22 @@ func (slamSvc *builtIn) getAndSaveDataDense(ctx context.Context, cams []camera.C
 
 // Creates a file for camera data with the specified sensor name and timestamp written into the filename.
 // For RGBD cameras, two filenames are created with the same timestamp in different directories.
-func createTimestampFilenames(cameraName, dataDirectory, fileType string, rgbd bool) []string {
+func createTimestampFilenames(cameraName, dataDirectory, fileType string, slamMode slam.Mode) []string {
 	// TODO change time format to .Format(time.RFC3339Nano) https://viam.atlassian.net/browse/DATA-277
 	timeStamp := time.Now()
 
-	if rgbd {
+	switch slamMode {
+	case slam.Rgbd:
 		colorFilename := filepath.Join(dataDirectory, "data", "rgb", cameraName+"_data_"+timeStamp.UTC().Format(slamTimeFormat)+fileType)
 		depthFilename := filepath.Join(dataDirectory, "data", "depth", cameraName+"_data_"+timeStamp.UTC().Format(slamTimeFormat)+fileType)
 		return []string{colorFilename, depthFilename}
+	case slam.Mono:
+		colorFilename := filepath.Join(dataDirectory, "data", "rgb", cameraName+"_data_"+timeStamp.UTC().Format(slamTimeFormat)+fileType)
+		return []string{colorFilename}
+	default:
+		filename := filepath.Join(dataDirectory, "data", cameraName+"_data_"+timeStamp.UTC().Format(slamTimeFormat)+fileType)
```

Comment:
[q] should the default be rgb? or is it better to leave this in data for when we add another server?

Reply:
I would suggest explicitly making a case for `Dim2d` (for Cartographer) and erroring if any other slam mode is passed in.

Reply:
yeah that makes sense, and we can add more cases when we support them